packages/chain-mon/src/balance-mon/service.ts

```diff
@@ -83,22 +102,40 @@ export class BalanceMonService extends BaseServiceV2<
           section: 'balances',
           name: 'getBalance',
         })
-        continue
       }
 
-      this.logger.info(`got balance`, {
-        address: account.address,
-        nickname: account.nickname,
-        balance: balance.toString(),
-      })
+      // Get the safe nonce to report
+      if (account.safe) {
+        let safeNonce: ethers.BigNumber
+        try {
+          safeNonce = BigNumber.from(
+            await this.options.rpc.call({
+              to: account.address,
+              data: '0xaffed0e0', // call the nonce() function in the safe contract
+            })
```

Comment:
Would it be better to use the Ethers contract wrapper instead of making a raw RPC call here?

Reply:
Definitely more readable to use an ethers contract I don't feel strongly myself though either way

Reply:
updated